### PR TITLE
AWS single and dualstack templates packages

### PR DESCRIPTION
**Pull request by igomez06**
**Background:**

- Implemented rke2 templates for both single and dual stack clusters
- Implemented aws terraform templates for single and dual stack enabled nodes
- New AWS package that will build RKE2 and Rancher packages based of off AWS that are both single and dual stack.

### CODE REVIEW

File: templates/rke2/overlay/opt/corral/rke2/init-cluster.sh
Comment (jakefhyde) [inline templates/rke2/overlay/opt/corral/rke2/init-cluster.sh:14]: Can this be removed?

File: templates/rancher/overlay/opt/corral/rancher/preflight.sh
Comment (jakefhyde) [inline templates/rancher/overlay/opt/corral/rancher/preflight.sh:6]: Could we potentially move this to it's own template?

File: templates/rancher/overlay/opt/corral/rancher/install-cert-manager.sh
Comment (jakefhyde) [inline templates/rancher/overlay/opt/corral/rancher/install-cert-manager.sh:16]: Why do we need to sleep here?

File: templates/aws/cluster_nodes/terraform/pools/main.tf
Comment (jakefhyde) [inline templates/aws/cluster_nodes/terraform/pools/main.tf:56]: Can we format all tf files?

File: templates/aws/cluster_nodes/manifest.yaml
Comment (jakefhyde) [inline templates/aws/cluster_nodes/manifest.yaml:8]: I think `optional` should be false by default.

**Review by jakefhyde — CHANGES_REQUESTED** [review]
Just a few comments, mostly nits.

File: templates/rke2/overlay/opt/corral/rke2/init-cluster.sh
Comment (igomez06) [reply templates/rke2/overlay/opt/corral/rke2/init-cluster.sh:14]: I removed it and moved it to preflight.sh, as ubuntu would need this pack to be able to use jq

File: templates/rancher/overlay/opt/corral/rancher/preflight.sh
Comment (igomez06) [reply templates/rancher/overlay/opt/corral/rancher/preflight.sh:6]: We could, but I think it's a nice to have kubectl installed when debugging or just view certain things in the cluster, but also it's not necessary as corral has functionality to get the kubeconfig file. It's your call, we can remove it altogether as well.

File: templates/rancher/overlay/opt/corral/rancher/install-cert-manager.sh
Comment (igomez06) [reply templates/rancher/overlay/opt/corral/rancher/install-cert-manager.sh:16]: I have issues when I attempt to install rancher right after installing cert-manager and the installation fails. But if I give it enough time, it will install with no issue.

File: templates/aws/cluster_nodes/manifest.yaml
Comment (igomez06) [reply templates/aws/cluster_nodes/manifest.yaml:8]: Fixed

File: templates/aws/cluster_nodes/terraform/pools/main.tf
Comment (igomez06) [reply templates/aws/cluster_nodes/terraform/pools/main.tf:56]: What do you mean by format?

**Comment by igomez06** [comment]
Jake could I get another review?

**Review by jakefhyde — APPROVED** [review]

**Comment by jakefhyde** [comment]
@igomez06 I think we are good to merge now, we can always update if needed.

**Comment by igomez06** [comment]
@jakefhyde Awesome! Thank you! Who has the permissions to merge?

**Comment by jakefhyde** [comment]
@igomez06 That would be me! Although I am looking for backups atm

**Pull request merged**
